review.py

```diff
@@ -0,0 +1,164 @@
+
+    def equip_item(self, item):
+        equippable = self.can_equip(item)
+        if item not in self.inventory.keys():
+            print("You don't have that item.")
+        if equippable is True:
+            slot = item.slot
+            equipment = self.equipped[slot]
+            if equipment is not None:
+                while True:
+                    inp = input(
+                        f"Would you like to replace {equipment} with {item}? y/n:\n")
+                    if inp == "y":
+                        self.adjust_stats(equipment, True)
+                    elif inp == "n":
+                        print("The item stays in your inventory.")
+                        break
+                    else:
+                        print("Please answer with \"y\" or \"n\".")
+                        continue
+            else:
+                self.equipped[slot] = item
+                del self.inventory[item]
+                self.adjust_stats(item)
+
+        else:
+            print(equippable)
+
+    def can_equip(self, item):
+        equippable = None
+        if item.function == "equip":
+            equippable = True
+        else:
+            equippable = "That item cannot be equipped."
+        if equippable is True and self.level >= item.level:
+            equippable = True
+        elif equippable is True:
+            equippable = "You are not a high enough level to equip that item."
+        return equippable
+
+    def display_stats(self):
+        print(f"{self.name}'s stats:")
+        print(f"Level: {self.level}")
+        print(f"Dream Level: {self.dream_level}")
+        print(f"Hit Points: {self.hp}")
+        print(f"Magic Points: {self.mp}")
+        print(f"Physical Attack: {self.phys_attack}")
+        print(f"Accuracy: {self.accuracy}")
+        print(f"Defense: {self.defense}")
+        print(f"Inventory:", end=" ")
+
+    def display_inventory(self):
+        inv_list = []
+        for key, value in self.inventory.items():
+            inv_list.append(f"{key.name.title()} x {value}")
+        if not inv_list:
+            print("You have nothing in your inventory.")
+        else:
+            for item in inv_list:
+                if inv_list[-1] == item:
+                    print(f"{item}.")
+                else:
+                    print(f"{item}, ", end="")
+
+    def display_equipped(self):
+        print("Equipped items:")
+        for key, value in self.equipped.items():
+            if value is None:
+                print(f"{key.title()}: Empty")
+            else:
+                print(f"{key.title()}: {value.name.title()}")
+
+
+class Item:
+    def __init__(self, name, description, location, function="use"):
+        self.name = name
+        self.description = description
+        self. location = location
+        self.function = function
+        self.equippable = False
+
+
+class Weapon(Item):
+    def __init__(self, name, description, location, function="equip",
```

Comment:
Functions with a bunch of parameters, particularly positional arguments, are sometimes easy to screw up. If you'd like, you can make parameters *forced* keyword arguments by putting a `*` in the parameter list. Everything after the bare `*` can then only be passed by keyword.

For example,

```
# Any of these can be passed by position or keyword
def __init__(self, name, description, location):

# location has to be passed by keyword
def __init__(self, name, description, *, location):

# everything has to be passed by keyword (self doesn't count)
def __init__(self, *, name, description, location):
```